src/lib/components/Codelab/index.js

```diff
@@ -48,16 +48,20 @@ class Codelab extends BaseElement {
     this._mql.removeListener(this._toggleDesktop);
   }
 
+  /**
+   * Normally LitElement will remove any light DOM children that are not
+   * slotted when we call render().
+   * Because we don't use slots, and we _do_ want to preserve this element's
+   * light DOM children (they hold the codelab instructions) we create a new
+   * renderRoot for LitElement.
+   * https://lit-element.polymer-project.org/guide/templates#renderroot
+   * This will render the glitch element as a sibling to the existing light
+   * DOM children.
+   */
   createRenderRoot() {
-    // Normally LitElement will remove any light DOM children that are not
-    // slotted when we call render().
-    // Because we don't use slots, and we _do_ want to preserve this element's
-    // light DOM children (they hold the codelab instructions) we create a new
-    // renderRoot for LitElement.
-    // https://lit-element.polymer-project.org/guide/templates#renderroot
-    // This will render the glitch element as a sibling to the existing light
-    // DOM children.
-    const container = document.createElement('div');
+    const container = /** @type this */ (
```

Comment:
Can you explain why `this` is needed?

Reply:
I think because of the fact we're extending from the `BaseElement` we end up getting this weird error:

```
Property 'createRenderRoot' in type 'Codelab' is not assignable to the same property in base type 'BaseElement'.
  Type '() => HTMLDivElement' is not assignable to type '() => this'.
```

So the casting was necessary in order to prevent errors.